src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -422,9 +423,16 @@ def schedule_task(self, task):
             #   - if the previous use included this node
             # If a tag exists, continue to consider this node if the tag was
             # used for this node - else continue to the next node.
-            if tag is not None and tag in self._tag_history:
-                if node_uid not in self._tag_history[tag]:
-                    continue
+            if tag is not None:
+                if tag in self._tag_history:
+                    if node_uid not in self._tag_history[tag]:
+                        continue
+                # for a new tag check that nodes were not used for previous tags
+                else:
+                    # `exclusive` -> not to share nodes between different tags
+                    is_exclusive = td.get('tags', {}).get('exclusive', True)
```

Comment:
I think this line should be:
```py
    is_exclusive = tag.get('exclusive', True)
```

Reply:
within the code `tag` is a string, but since `'tags'` is in scheme, then it always will be in task description -> `is_exclusive = td['tags'].get('exclusive', True)`. 

Or you want move it inside `'colocate'`? (`td = {..'tags': {'colocate': {'name': <tag_name>, 'exclusive': False}}..}`) I would like to keep it as a separate key within `tags`

Reply:
We have in line 383:
```py
tag = td.get('tag')
```
and thus the first part of 
```py
td.get('tags', {}).get('exclusive', True)
```
kind of duplicates it.  But you are right - the former one would need a dict default value then.

Anyway, not important, this comment was not meant to mean much more than this :-)

Reply:
right, that is `tag` (of `str` type), not `tags` :) and that `tag` attribute was temporary and is close to become obsolete